File: rfi-frontend/src/types/News.ts
export interface NewsArticle {
  id: number;
  title: string;
  slug: string;
  summary: string;
  category: string | null;
  date: string;
  readTimeMinutes: number | null;
  videoUrl: string | null;
  imageUrl: string | null;
  createdAt: string;
  sections: NewsSection[];
}

export interface NewsSection {
  id: number;
  sectionType: string;
  title: string | null;
  orderIndex: number;
  keyPoints: KeyPoint[];
}

export interface KeyPoint {
  id: number;
  title: string | null;
  description: string;
  orderIndex: number;
}

File: RFI.API/Models/News.cs
namespace RFI.API.Models
{
    public class News
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Slug { get; set; } = string.Empty;
        public string Summary { get; set; } = string.Empty;
        public string? Category { get; set; }
        public DateTime Date { get; set; }
        public int? ReadTimeMinutes { get; set; }
        public string? VideoUrl { get; set; }
        public string? ImageUrl { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        // Navigation properties
        public ICollection<NewsSection> Sections { get; set; } = new List<NewsSection>();
    }
}